include/alpaka/core/Common.hpp

```diff
@@ -150,25 +184,40 @@
 //! because this is forbidden by CUDA.
 //!
 //! \attention It is not allowed to initialize the variable together with the declaration.
-//!            To initialize the variable alpaka::createStaticDevMemView and alpaka::memcpy must be used.
+//!            To initialize the variable alpaka::memcpy must be used.
 //! \code{.cpp}
-//! ALPAKA_STATIC_ACC_MEM_CONSTANT int foo;
+//! ALPAKA_STATIC_ACC_MEM_CONSTANT(int, foo);
+//!
+//! struct DeviceMemoryKernel
+//! {
+//!    ALPAKA_NO_HOST_ACC_WARNING
+//!    template<typename TAcc>
+//!    ALPAKA_FN_ACC void operator()(TAcc const& acc) const
+//!    {
+//!      auto a = foo.get();
+//!    }
+//!  }
 //!
 //! void initFoo() {
 //!     auto extent = alpaka::Vec<alpaka::DimInt<1u>, size_t>{1};
-//!     auto viewFoo = alpaka::createStaticDevMemView(&foo, device, extent);
 //!     int initialValue = 42;
 //!     alpaka::ViewPlainPtr<DevHost, int, alpaka::DimInt<1u>, size_t> bufHost(&initialValue, devHost, extent);
-//!     alpaka::memcpy(queue, viewGlobalMemUninitialized, bufHost, extent);
+//!     alpaka::memcpy(queue, foo, bufHost, extent);
 //! }
 //! \endcode
 #if((BOOST_LANG_CUDA && BOOST_COMP_CLANG_CUDA) || (BOOST_LANG_CUDA && BOOST_COMP_NVCC && BOOST_ARCH_PTX)              \
     || BOOST_LANG_HIP)
-#    define ALPAKA_STATIC_ACC_MEM_CONSTANT __constant__
+#    define ALPAKA_STATIC_ACC_MEM_CONSTANT(type, name)                                                                \
+        template<typename TAcc>                                                                                       \
+        inline __constant__ alpaka::DevGlobal<type> name
```

Comment:
would
```suggestion
        inline __constant__ alpaka::DevGlobal<const type> name
```
work ?

(also for the CPU case)

Reply:
From a quick try, I'd say no because `T value` inside the `struct DevGlobal` is mapped to `const type value` that then needs to be initialized when the variable is declared

Reply:
Maybe you could store `std::remove_const_t<T>` as a data member (so `memcpy` can write to it), but return a `T&` or `T*` (including the `const`) in `T& get()` and `T* operator&()` ?

Reply:
do you mean
```cpp
    template<typename T>
    struct DevGlobal
    {
        using Type = std::remove_const_t<T>;
        Type value; // backend specific value
    }
```
?
The `memcpy` sees `T`, not `Type`, and fails :(

Reply:
Some options:
  - add a `private` non-`const` `get()` method, and make the function that calls `memcpy` a `friend` of `DevGlobal` so it can use it
  - use `const_cast<std::remove_const_t<T>*>()` in the call to `memcpy` (this should be safe because the underlying `value` is not `const`)
  - use two different objects, one with a `const` `get()` on the device and one with a non-`const` `get()` on the host (this is just a random idea, not sure it's actually feasible)

Reply:
I went for the second option